src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/FunctionPointerUnmanagedCallingConventionClassifier.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.Classification.Classifiers;
+using Microsoft.CodeAnalysis.Collections;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.Classification;
+
+internal sealed class FunctionPointerUnmanagedCallingConventionClassifier : AbstractSyntaxClassifier
+{
+    public override ImmutableArray<Type> SyntaxNodeTypes { get; } = ImmutableArray.Create(typeof(FunctionPointerUnmanagedCallingConventionSyntax));
+
+    public override void AddClassifications(
+        SyntaxNode syntax,
+        TextSpan textSpan,
+        SemanticModel semanticModel,
+        ClassificationOptions options,
+        SegmentedList<ClassifiedSpan> result,
+        CancellationToken cancellationToken)
+    {
+        var callingConvention = (FunctionPointerUnmanagedCallingConventionSyntax)syntax;
+        var name = callingConvention.Name.ValueText;
```

Comment:
This be an empty string, e.g. in the case of a missing identifier inside the brackets, right? In that case there's nothing to classify. Should this be special-cased with an early return so that we don't have to do the (redundant) type lookup?

Reply:
It doesn't seem like an important micro-optimization to have. Thanks though!